cppcodec/detail/stream_codec.hpp

```diff
@@ -241,7 +241,7 @@ struct alphabet_index_info
     static constexpr const alphabet_index_t padding_idx = 1 << 8;
     static constexpr const alphabet_index_t invalid_idx = 1 << 9;
     static constexpr const alphabet_index_t eof_idx = 1 << 10;
-    static constexpr const alphabet_index_t stop_character_mask = ~0xFF;
+    static constexpr const alphabet_index_t stop_character_mask = ~alphabet_index_t{0xFF};
```

Comment:
Shouldn't this be the other way round? As in, `alphabet_index_t{~0xFF}` (which not only ensures that the cast is the last operation before assigning, but is also easier to read imho)?

Reply:
Nope... 

MSVC: ` error C2397: conversion from 'int' to 'cppcodec::detail::alphabet_index_t' requires a narrowing conversion`
GCC: ` error: narrowing conversion of ‘-256’ from ‘int’ to ‘cppcodec::detail::alphabet_index_t {aka long unsigned int}’ inside { } [-Wnarrowing]
     static constexpr const alphabet_index_t stop_character_mask = alphabet_index_t{~0xFF};`

Reply:
`static_cast<alphabet_index_t>(~0xFF)` would probably work, if you prefer that

Reply:
Ah, this is interesting. `alphabet_index_t` is defined as `uint_fast16_t` so depending on how wide it actually is, some bytes may get cut off by a cast. For the warning (error) that you listed, it would probably be enough to write it as `0xFFu` to make the narrowing go away, but I feel like that wouldn't do it if it were actually just 16 bits wide.

Maybe the best course of action is to write it as 0xFF00, get rid of the complement altogether, and make sure that the benchmarks don't deteriorate. (It should be fine but I've had some surprises in this area before, so I want to look out for potential issues here.)